### DISCUSSION TITLE
Sharepoint file versions

### DISCUSSION THREAD

**Discussion by Ranjith-2024** [Barbarur/NovaPoint#40] [Q&A]
Hello Barbarur ,

Thanks for your reply on this below issue & Hope you are well..Is there any videos / documents avaibale for to use this app you mentioned ? 

We have multiple SPO sites impacted with huge amount of storage currently. Planned to delete the old file versions from each sites and to keep only the updated file version. Some of the sites are having 400 gb to 1200 GB of storage. (SPO Currently Ruuning out of storage space) We need to know the current file versions being used in each sites , so that it will hep us to know the existing file versions of each file / document per site and can decide which versions to be used and delete all other old versions. (We are working with MS)

We need a PnP Script to fetch above requirement / data/ report from each SPO Sites , Once we have the reports ready from each sites , decision will be taken accordingly for the deletions of Old versions. We may need a script at that point of time for deleting old versions from each sites as well.

Could you please have a look and help us with Scripts asap. OPened a case with PnP , no reply has been received yet. Really Appreciate your contributions..

Regards
Ranjith

**Comment by Barbarur** [comment]
Hi @Ranjith-2024 ,

NovaPoint is an app rather than a script, you need to download and install on your device; This is a video introduction of the app: https://youtu.be/-0__y4mBYto?si=SbfZ6s3YADwZ5sCL

Here is how to setup the app: https://github.com/Barbarur/NovaPoint/wiki/Setup

I don't have a exact video the delete versions, but here is a sample video of how to use the app: https://youtu.be/m3nDaHi9Ygw?si=tddaqUQHwhNWGclQ

You can start by running a report of all the files, which includes the number of versions and the total size consumed by the versions: https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ItemReport

Then you can delete the file versions: https://github.com/Barbarur/NovaPoint/wiki/Solution-Automation-RemoveFileVersionAuto

The app is quite straight forward to us. You can install and play around. Most of the operation that make any impact on the environment, like the deletion, have a "report mode" you can use to have an idea of the impact before performing the change.

I hope it helps.

**Comment by TeMi-SHP** [comment]
Hi,

I would like to ask if there is a tool or script that can help with the following tasks, considering we have 62,000 SharePoint Online sites:

1) Extract the size of the Preservation Hold Library (PHL) from Storage Metrics without accessing each PHL and calculating the size of individual items.
2) Generate an export for all 62,000 sites, listing the number of versions for each document and the number of versions per document in each document library across all sites.

Is there a way to obtain such exports?

Thank you!

**Comment by Barbarur** [comment]
Hi @TeMi-SHP ,

For the first report you can use the [Libraries and Lists report](https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ListReport), select **Single library** or list and add the ***Preservation Hold Library*** as the title. Make sure you select **Include list size**

![image](https://github.com/user-attachments/assets/5677b127-d5b0-492d-94b7-acac826cac30)

For the second report you can us [Files and Items report](https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ItemReport). It wil generate a report of all the files including size and total size (including versions). Though for 62,000 sites it would take too long to run. I recommend to run either [Site scollection and subsites report](https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ItemReport) or [Libraries and Lists report](https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ListReport). You can see the Sites/Libraries that consume the most storage, then run [Files and Items report](https://github.com/Barbarur/NovaPoint/wiki/Solution-Report-ItemReport) by batches of 10-20 sites, depending on the site, by selecting the option **List of Site URLs**.

**Comment by TeMi-SHP** [comment]
Hi Barbour,

Thank you for your answer. Would it be possible to get in touch via Teams, for example, to discuss your app and also see how it works? I work for a large corporate company with many restrictions, so before implementing anything, I need to be sure that it is worthwhile for us.

I would really appreciate it if you could spare 30 minutes to discuss your app with me. Please let me know. Thank you.

**Comment by Barbarur** [comment]
Hi @TeMi-SHP ,

You can reach me at ***novapointapp@protonmail.com***
